principles.rst

```diff
@@ -158,32 +159,108 @@ there is an absence of punning.
+
+- Positive-position signatures' free vars cause  implicit ``forall ... .``
+
+- Negative position free vars cause different sorts of binding:
+
+  - Signatures on term patterns (pattern signatures) cause implicit ``let type ... = _ in``
+
+  - Signatures on type variables (kind signatures) case implicit ``@...``
+
+From `#425`_, `#448`_.
+
 Lexical Scoping Principle (LSP)
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 .. _`Lexical Scoping Principle`:
 
 **Principle**:
 
-a. For every appearance of
-an identifier, it is possible to determine whether that appearance is a *binding site*
-or an *occurrence* without examining the context.
+a. For every appearance of an identifier, it is possible to determine whether that appearance is
+   a mere *occurrence*, and thus must be bound elsewhere for the program to be valid.
+   or the variable is a *binding site* (or causes an implicit binding site, basically the same thing).
+
+without examining the context.
 
 b. For every *occurrence* of an
 identifier, it is possible to uniquely identify its *binding site*, without
 involving the type system.
 
+This builds upon the `Explicit Binding Principle`_:
+whereas that former principle ensures that explicit alternatives to implicit binding constructs *exist at all*,
+this latter principle makes those explicit alternatives *compulsory*, because we must not have implicit binding in order to uphold this principle.
+
 The `Lexical Scoping Principle`_ is almost true today, with the following nuances:
 
-1. Template Haskell splices may need to be run before completing name resolution (and running those splices requires type-checking them).
+#. Template Haskell splices may need to be run before completing name resolution (and running those splices requires type-checking them).
+
+#. The `deprecated mechanism <https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/exts/duplicate_record_fields.html#selector-functions>`_ for disambiguating duplicate record fields violates the `Lexical Scoping Principle`_ by requiring the type system.
+
+#. In a pattern signature, if we have ``f (x :: Maybe a)``, the ``a``
+   is an occurrence if ``a`` is already in scope, and is implicitly bound otherwise.
 
-2. The `deprecated mechanism <https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/exts/duplicate_record_fields.html#selector-functions>`_ for disambiguating duplicate record fields violates the `Lexical Scoping Principle`_ by requiring the type system.
+#. In a type signature, if we have ``f :: a -> a``, the ``a``
+   is an occurrence if ``a`` is already in scope, and it implicitly bound otherwise.
 
-3. In a pattern signature, if we have ``f (x :: Maybe a)``, the ``a``
-   is an occurrence if ``a`` is already in scope, and it is a binding site otherwise.
+#. In a kind signature, if we have ``data Foo (a :: k)``, the ``k``
```

Comment:
do you maybe want to include a code example with `data Foo @k (a :: k)` here?

Reply:
I don't remember what my intention was here, better disregard my comment

Reply:
ah, it was because of "historically impossible", to illustrate how it would be possible – but this has already been demonstrated above

Reply:
I removed historically impossible because even more `@k` there is plain `k`: `data Foo k (a :: k)` has worked since 8.0! No need worrying about things older than that at this point. I don't think.